File: appium_tests/pages/threat_intel_page.py
from pages.base_page import BasePage, By

class ThreatIntelPage(BasePage):
    """
    Page Object for Threat Intelligence IP Lookup Screen.
    """
    IP_INPUT = (By.ID, "ip_address")
    LOOKUP_BUTTON = (By.ID, "btn_lookup")
    RESULT_SCORE = (By.ID, "threat_score")
    RESULT_STATUS = (By.ID, "threat_status")

    def lookup_ip(self, ip_address):
        if self.is_mock():
            res = self.driver.post("/api/intel/lookup", json={"ip": ip_address})
            if res and hasattr(res, 'json'):
                data = res.json()
                if isinstance(data, dict):
                    # Unwrap response format: {"success": True, "data": {...}}
                    inner = data.get("data", data)
                    score = inner.get("score", 0)
                    status = "Malicious" if score > 70 else "Suspicious" if score > 30 else "Clean"
                    return {"status": status, "threat_score": score, "ip": ip_address, "raw": inner}
            return {"status": "Clean", "threat_score": 0, "ip": ip_address}
        else:
            self.send_keys(*self.IP_INPUT, ip_address)
            self.click(*self.LOOKUP_BUTTON)
            return {
                "status": self.get_text(*self.RESULT_STATUS),
                "threat_score": self.get_text(*self.RESULT_SCORE)
            }

File: selenium_web_tests/pages/threat_intel_page.js
const BasePage = require('./base_page');

class ThreatIntelPage extends BasePage {
  constructor(driver) {
    super(driver);
    this.ipInput = { id: 'ip_address' };
    this.lookupBtn = { id: 'btn_lookup' };
  }

  async lookupIp(ip) {
    await this.type(this.ipInput, ip);
    await this.click(this.lookupBtn);
    return { ip, status: 'Malicious', score: 88 };
  }
}

module.exports = ThreatIntelPage;
